src/node.cc

```diff
@@ -2579,18 +2579,34 @@ void Init(std::vector<std::string>* argv,
     // [0] is expected to be the program name, fill it in from the real argv.
     env_argv.push_back(argv->at(0));
 
-    // Split NODE_OPTIONS at each ' ' character.
-    std::string::size_type index = std::string::npos;
-    do {
-      std::string::size_type prev_index = index;
-      index = node_options.find(' ', index + 1);
-      if (index - prev_index == 1) continue;
-
-      const std::string option = node_options.substr(
-          prev_index + 1, index - prev_index - 1);
-      if (!option.empty())
-        env_argv.emplace_back(std::move(option));
-    } while (index != std::string::npos);
+    bool will_start_new_arg = true;
+    for (std::string::size_type index = 0;
+         index < node_options.size();
+         ++index) {
+      char c = node_options.at(index);
+
+      // Backslashes escape the following character
+      if (c == '\\') {
+        if (index + 1 == node_options.size()) {
+          fprintf(stderr,
+                  "%s: invalid escaping for NODE_OPTIONS\n",
+                  argv->at(0).c_str());
+          exit(9);
```

Comment:
Is this really OK? Will it mess up embedders, @codebytere @rubys , or are they already in trouble and this doesn't make it any worse?

Reply:
i believe this'd be fine on our end!

Reply:
Would you like a different exit code? I put `9` because it was what was below, but I can change it

Reply:
`9` is appropriate: https://nodejs.org/dist/latest-v11.x/docs/api/process.html

> * `9` - **Invalid Argument** - Either an unknown option was specified, or an option requiring a value was provided without a value.

Reply:
@sam-github I would expect embedders to use `argc`, `argv` and pretty much avoid `NODE_OPTIONS`.

Reply:
@arcanis The situation here has actually changed since this PR was opened – we don’t want to `exit()` or print to stderr here now, but rather push the error into `errors` and return the exit code instead

Reply:
@addaleax Fixed 👍